### PR TITLE
[RLlib] - Enable multi-learner setup for hybrid stack BC

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

This PR does fix a bug in hybrid stack BC which was hindering multi-learner setups for offline RL.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by sven1977 — APPROVED** [review]
LGTM! Thanks for this fix @simonsays1980 .

File: rllib/examples/offline_rl/pretrain_bc_single_agent_evaluate_as_multi_agent.py [hunk 1]
```diff
@@ -147,6 +147,7 @@
         TRAINING_ITERATION: args.stop_iters,
     }
 
+    args.local_mode = True
```
Comment (sven1977) [inline rllib/examples/offline_rl/pretrain_bc_single_agent_evaluate_as_multi_agent.py:150]: remove?

File: rllib/policy/sample_batch.py [hunk 2]
```diff
@@ -907,14 +908,20 @@ def get(self, key, default=None):
             return default
 
     @PublicAPI
-    def as_multi_agent(self) -> "MultiAgentBatch":
-        """Returns the respective MultiAgentBatch using DEFAULT_POLICY_ID.
+    def as_multi_agent(self, module_id: Optional[ModuleID] = None) -> "MultiAgentBatch":
```
Comment (sven1977) [inline rllib/policy/sample_batch.py:911]: perfect!

File: rllib/examples/offline_rl/pretrain_bc_single_agent_evaluate_as_multi_agent.py (hunk 1 above)
Comment (simonsays1980) [reply rllib/examples/offline_rl/pretrain_bc_single_agent_evaluate_as_multi_agent.py]: Good catch!

**Pull request merged**